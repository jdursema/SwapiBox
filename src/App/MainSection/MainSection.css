.button{
  font-size: 20px;
  width: 150px;
  border-radius: 0;
  border: none;
  margin: 1%;
  padding: 1% 0;
}



.btn-bar{
  display: flex;
  flex-direction: row;
  justify-content: center;
}


.main-section{
  display: flex;
  flex-direction: column;
  align-items: center;
}

.Planets{
  background-color: lightblue;
}

.Vehicles{
  background-color: lightgreen;
}

.Planets{
  background-color: purple;
}