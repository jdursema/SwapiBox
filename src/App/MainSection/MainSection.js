import React from 'react'
import './MainSection.css'
import CardContainer from './CardContainer/CardContainer'

const MainSection = ({fetchCharacterCardInfo, fetchVehicleCardInfo, fetchPlanetCardInfo, cardsInfo}) => {
  return (
    <div className='main-section'>
      <div className= 'btn-bar'>
        <button className='People button' onClick={()=>{fetchCharacterCardInfo()}}>People</button>
        <button className='Vehicles button' onClick={()=>{fetchVehicleCardInfo()}}>Vehicles</button>
        <button className='Planets button' onClick={()=>{fetchPlanetCardInfo()}}>Planets</button>

      </div>
      <CardContainer cardsInfo = {cardsInfo} />
    </div>
  )
}

export default MainSection