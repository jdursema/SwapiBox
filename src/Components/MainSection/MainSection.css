.button {
  font-size: 20px;
  width: 225px;
  border-radius: 10px;
  border: none;
  margin: 2%;
  padding: 1% 0;
  background-color: #3F3F3F;
  font-size: 35px;
  font-family: 'Pathway Gothic One', sans-serif;
  color: #FEFEFE; }

.btn-bar {
  display: flex;
  flex-direction: row;
  justify-content: center;
  width: 70%; }

.main-section {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  margin: 0 5%; }

.People:hover {
  animation: pulse 5s infinite; }

@keyframes pulse {
  0% {
    background-color: white; }
  50% {
    background-color: #76D7F2;
    color: white; }
  100% {
    background-color: white; } }

.People:focus {
  background-color: #76D7F2;
  color: white; }

.Vehicles:hover {
  animation: pulse2 5s infinite; }

@keyframes pulse2 {
  0% {
    background-color: white; }
  50% {
    background-color: #27FF62;
    color: white; }
  100% {
    background-color: white; } }

.Vehicles:focus {
  background-color: #27FF62;
  color: white; }

.Planets:hover {
  animation: pulse3 5s infinite; }

@keyframes pulse3 {
  0% {
    background-color: white; }
  50% {
    background-color: #8000FF;
    color: white; }
  100% {
    background-color: white; } }

.Planets:focus {
  background-color: #8000FF;
  color: white; }
