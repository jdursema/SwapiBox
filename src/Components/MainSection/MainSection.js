import React from 'react';
import './MainSection.css';
import CardContainer from '../CardContainer/CardContainer';
import PropTypes from 'prop-types';

const MainSection = ({ fetchCharacterCardInfo, fetchVehicleCardInfo, fetchPlanetCardInfo, cardsInfo, addToFavorites, Favorites }) => {
  return (
    <div className='main-section'>
      <div className='btn-bar'>
        <button 
          className='People button' 
          onClick={() => { fetchCharacterCardInfo(); } }>People
        </button>
        <button 
          className='Vehicles button' 
          onClick={() => { fetchVehicleCardInfo(); } }>Vehicles
        </button>
        <button
          className='Planets button' 
          onClick={() => { fetchPlanetCardInfo(); } }>Planets</button>
      </div>

      <CardContainer 
        cardsInfo={cardsInfo} 
        addToFavorites={addToFavorites}
        Favorites={Favorites} />
    </div>
  );
};

export default MainSection;

MainSection.propTypes = {
  fetchCharacterCardInfo: PropTypes.func,
  fetchVehicleCardInfo: PropTypes.func,
  fetchPlanetCardInfo: PropTypes.func,
  cardsInfo: PropTypes.array,
  addToFavorites: PropTypes.func,
  Favorites: PropTypes.array
};