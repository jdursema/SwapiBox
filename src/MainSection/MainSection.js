import React from 'react'
import CardContainer from '../CardContainer/CardContainer'

const MainSection = () => {
  return (
    <div>
      <p>This will be the main section</p>
      <CardContainer />
    </div>
  )
}

export default MainSection