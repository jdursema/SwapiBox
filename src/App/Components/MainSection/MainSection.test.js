import React from 'react';
import { shallow } from 'enzyme';
import MainSection from './MainSection';


it('should render without crashing', () => {
  const mainSectionWrapper = shallow(<MainSection />);

  expect(mainSectionWrapper).toBeDefined();
});

it('should match the snapshot', () => {
  const mockFunc = jest.fn();
  const mockFunc2 = jest.fn();
  const mockFunc3 = jest.fn();
  const mockFunc4 = jest.fn();
  const cardData = 
    [{name: 'Julie', Data:{Homeworld: 'earth'}},
      {name: 'Pluto', Data:{Residents: 'Goofy'}}];

  const mainSectionWrapper = shallow(<MainSection 
    fetchCharacterCardInfo={mockFunc} 
    fetchVehicleCardInfo={mockFunc2} 
    fetchPlanetCardInfo={mockFunc3} 
    cardsInfo={cardData} 
    addToFavorites={mockFunc4} />);

  expect(mainSectionWrapper).toMatchSnapshot();
});
