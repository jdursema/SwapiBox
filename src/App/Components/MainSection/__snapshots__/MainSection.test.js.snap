// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainSection
    addToFavorites={[Function]}
    cardsInfo={
        Array [
            Object {
              "data": Object {
                "Homeworld": "earth",
              },
              "name": "Julie",
            },
            Object {
              "data": Object {
                "Residents": "Goofy",
              },
              "name": "Pluto",
            },
          ]
    }
    fetchCharacterCardInfo={[Function]}
    fetchPlanetCardInfo={[Function]}
    fetchVehicleCardInfo={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="btn-bar"
>
          <button
                    className="People button"
                    onClick={[Function]}
          >
                    People
          </button>
          <button
                    className="Vehicles button"
                    onClick={[Function]}
          >
                    Vehicles
          </button>
          <button
                    className="Planets button"
                    onClick={[Function]}
          >
                    Planets
          </button>
</div>,
        <CardContainer
          Favorites={undefined}
          addToFavorites={[Function]}
          cardsInfo={
                    Array [
                              Object {
                                "data": Object {
                                  "Homeworld": "earth",
                                },
                                "name": "Julie",
                              },
                              Object {
                                "data": Object {
                                  "Residents": "Goofy",
                                },
                                "name": "Pluto",
                              },
                            ]
          }
/>,
      ],
      "className": "main-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="People button"
              onClick={[Function]}
>
              People
</button>,
            <button
              className="Vehicles button"
              onClick={[Function]}
>
              Vehicles
</button>,
            <button
              className="Planets button"
              onClick={[Function]}
>
              Planets
</button>,
          ],
          "className": "btn-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "People",
              "className": "People button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "People",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vehicles",
              "className": "Vehicles button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Planets",
              "className": "Planets button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Planets",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "Favorites": undefined,
          "addToFavorites": [Function],
          "cardsInfo": Array [
            Object {
              "data": Object {
                "Homeworld": "earth",
              },
              "name": "Julie",
            },
            Object {
              "data": Object {
                "Residents": "Goofy",
              },
              "name": "Pluto",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="btn-bar"
>
            <button
                        className="People button"
                        onClick={[Function]}
            >
                        People
            </button>
            <button
                        className="Vehicles button"
                        onClick={[Function]}
            >
                        Vehicles
            </button>
            <button
                        className="Planets button"
                        onClick={[Function]}
            >
                        Planets
            </button>
</div>,
          <CardContainer
            Favorites={undefined}
            addToFavorites={[Function]}
            cardsInfo={
                        Array [
                                    Object {
                                      "data": Object {
                                        "Homeworld": "earth",
                                      },
                                      "name": "Julie",
                                    },
                                    Object {
                                      "data": Object {
                                        "Residents": "Goofy",
                                      },
                                      "name": "Pluto",
                                    },
                                  ]
            }
/>,
        ],
        "className": "main-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="People button"
                onClick={[Function]}
>
                People
</button>,
              <button
                className="Vehicles button"
                onClick={[Function]}
>
                Vehicles
</button>,
              <button
                className="Planets button"
                onClick={[Function]}
>
                Planets
</button>,
            ],
            "className": "btn-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "People",
                "className": "People button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "People",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Vehicles",
                "className": "Vehicles button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Planets",
                "className": "Planets button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Planets",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "Favorites": undefined,
            "addToFavorites": [Function],
            "cardsInfo": Array [
              Object {
                "data": Object {
                  "Homeworld": "earth",
                },
                "name": "Julie",
              },
              Object {
                "data": Object {
                  "Residents": "Goofy",
                },
                "name": "Pluto",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
